src/Features/tutorial.ts

```diff
@@ -0,0 +1,117 @@
+import type { HasPlayerMovedEvent } from "@workadventure/iframe-api-typings/Api/Events/HasPlayerMovedEvent";
+import type { WasCameraUpdatedEvent } from "../../../workadventure/front/src/Api/Events/WasCameraUpdatedEvent";
+import { desktopConfig, mobileConfig } from "../Iframes/Tutorial/config/config";
+
+export async function initTutorial(): Promise<void> {
+    const tutorialDone = WA.player.state.tutorialDone;
+
+    if (!tutorialDone) {
+        openTutorial();
+
+        let playerPosition: HasPlayerMovedEvent = await WA.player.getPosition();
+        let camera: WasCameraUpdatedEvent;
+
+        const tutorialIFrame = await WA.room.website.get("tutorial");
+
+        const updatePosition = (): void => {
+            // We do not want the iFrame to cross the worldView's limit, as it could become invisible for the player
+            const margin = 16;
+            const rightBorderCrossed =
+                playerPosition.x + tutorialIFrame.x + tutorialIFrame.width >
+                camera.x + camera.width;
+            const leftBorderCrossed = playerPosition.x + tutorialIFrame.x < camera.x;
+            const topBorderCrossed =
+                playerPosition.y + tutorialIFrame.y + tutorialIFrame.height >
+                camera.y + camera.height;
+            const bottomBorderCrossed = playerPosition.y + tutorialIFrame.y < camera.y;
+
+            if (rightBorderCrossed) {
+                tutorialIFrame.x = -tutorialIFrame.width - 1.5 * margin;
+            } else if (leftBorderCrossed) {
+                tutorialIFrame.x = 1.5 * margin;
+            }
+
+            if (topBorderCrossed) {
+                tutorialIFrame.y = -tutorialIFrame.height;
+            } else if (bottomBorderCrossed) {
+                tutorialIFrame.y = margin;
+            }
+        };
+
+        const processIframeConfig = (config: {
+            width: number;
+            height: number;
+            scale: number;
+        }): void => {
+            tutorialIFrame.width = config.width;
+            tutorialIFrame.height = config.height;
+            tutorialIFrame.scale = config.scale;
+        };
+
+        const updateProportions = (zoomLevel: number): void => {
+            // If the zoom level is too high and our iFrame can't fit into the worldView, want to adapt its format
+            console.log(zoomLevel);
+
+            const config = /Mobi|Android/i.test(navigator.userAgent) ? mobileConfig : desktopConfig;
```

Comment:
You should move the Regex `/Mobi|Android/i.test(navigator.userAgent)` to the top file scope.

The result of the regex is always the same, so there is no need to recompute the regex each time the camera is updated.